### PR TITLE
Added check for [Test] attribute on all public methods in a [TestFixture]

### PR DESCRIPTION
**Pull request by jbheard**

### CODE REVIEW

**Comment by j3parker** [comment]
It looks like there are still some whitespace issues. I'll come over and we can see if we need to poke at our code style settings.

**Comment by j3parker** [comment]
I think there was some desire to report errors when we had `[Test]`s in non-`[TestFixture]`s... (maybe that gets hard with those crummy test base classes?) so the control flow would need to change a bit for that, but we can do that later. @travisMiehm thoughts?

**Comment by travisMiehm** [comment]
I think test attributes in non fixtures can be a next step, and the case covered by this change is good.

File: src/D2L.CodeStyle.TestAnalyzers/Common/Diagnostics.cs [hunk 1]
```diff
@@ -51,5 +51,15 @@ public static class Diagnostics {
 			isEnabledByDefault: true,
 			description: "Custom service locators are expensive to instantiate and slow down tests significantly. Use the default locator instead."
 		);
-	}
+
+        public static readonly DiagnosticDescriptor TestAttributeMissed = new DiagnosticDescriptor(
+            id: "D2LTESTS006",
+            title: "Method not labelled as [Test], [TestCase], or [TestCaseSource]",
```
Comment (travisMiehm) [inline src/D2L.CodeStyle.TestAnalyzers/Common/Diagnostics.cs]: Ah, sorry, we also have some tests that only use `[Theory]`.

I've skimmed [NUnit's list of attributes](https://github.com/nunit/docs/wiki/Attributes), and I think this is the last "hey, this method is a test" attribute that we use.

Comment (jbheard) [reply src/D2L.CodeStyle.TestAnalyzers/Common/Diagnostics.cs]: Added the theory attribute

**Comment by j3parker** [comment]
Ok, we need to wittle down the in-flight analyzer changes :) Let's focus on getting the LMS up-to-snuff now. We can't merge the analyzers until it's ready.

**Comment by j3parker** [comment]
You could make your new diagnostics be info if you want.

**Pull request merged**